### DISCUSSION TITLE
Updating map

### DISCUSSION THREAD

**Discussion by mattiasmar** [introlab/rtabmap#1093] [Q&A] [answered]
Is there an API to re-map parts of a map? 
I want to unfix a part of the 3D reconstruction and its associated keyframes (signatures) when I've detected that a certain area has changed. Is there an established method in rtabmap for doing this?

**Comment by matlabbe** [comment, marked as answer]
Currently, to "append" a new session, we can set parameter `Rtabmap/StartNewMapOnLoopClosure` to true so that a new session will start only when the robot/camera is first localized in the previous map. That way, we make sure that the new session is merged with previous one(s). Note if you remap the same environment, the old nodes of same location will still be in the map. If you want to remove those old nodes, there is no automatic tool to do this. 

One manual way to remove nodes in the map is with `rtabmap-databaseViewer`, with Constraints view and Graph View. In Graph view, search for the node or group of nodes you want to remove. The idea is that with Constraint view, you search the links that link those nodes to remaining graph, then click Reject to remove them. Once all links linking a node to graph are removed, the node should disappear from the graph.

Other approaches rely on editing the sqlite3 database directly, though one should carefully remove links in Link table, nodes in Node/Data tables and corresponding features in Feature table. `opt_ids` and `opt_poses` form Admin table would also need to be set to null (if they exist) to force rtabmap to re-optimize the graph with remaining nodes.
